### PR TITLE
MGMT-2966: add a single node mode for installing SNO clusters

### PR DESCRIPTION
**Pull request by osherdp**

### CODE REVIEW

**Review by eranco74 — COMMENTED** [review]
Looks good, 
Add tests

**Comment by osherdp** [comment]
> Looks good,
> Add tests

I thought about adding tests in the next issue, where the profile have some more complete logic
Right now it can be a simple test that just changes the model. But tell me if you think it's interesting enough I can add this kind of test for now

**Comment by avishayt** [comment]
I prefer one PR (with several commits) that includes the API change, generated code, as well as the actual implementation.

File: swagger.yaml [hunk 1]
```diff
@@ -2867,6 +2867,13 @@ definitions:
         type: string
         description: OpenShift cluster name.
         x-nullable: true
+      mode:
```
Comment (avishayt) [inline swagger.yaml:2870]: What other enum values do we envision in the future?
Either we should change "mode" to something more specific like "resiliency", and in the future, we can add active-passive, or maybe we should leave it as "mode" if we want to add other non-resiliency values.  I'm guessing the first option.

Comment (eranco74) [reply swagger.yaml:2870]: This should match with https://github.com/openshift/api/pull/827/files#diff-3ff94e87467e8743ff79a40fa85f123d3205a560f79b4dea1919e76a4d55958dR181

Comment (osherdp) [reply swagger.yaml]: done

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Pull request merged**